core/tx_journal.go

```diff
@@ -76,33 +100,57 @@ func (journal *txJournal) load(add func([]*types.Transaction) []error) error {
 	// Inject all transactions from the journal into the pool
 	stream := rlp.NewStream(input, 0)
 	total, dropped := 0, 0
+	batch := types.PoolTransactions{}
 
 	// Create a method to load a limited batch of transactions and bump the
 	// appropriate progress counters. Then use this method to load all the
 	// journaled transactions in small-ish batches.
-	loadBatch := func(txs types.Transactions) {
+	loadBatch := func(txs types.PoolTransactions) {
 		for _, err := range add(txs) {
 			if err != nil {
 				utils.Logger().Error().Err(err).Msg("Failed to add journaled transaction")
 				dropped++
 			}
 		}
 	}
-	var (
-		failure error
-		batch   types.Transactions
-	)
 	for {
-		// Parse the next transaction and terminate on error
-		tx := new(types.Transaction)
+		// Parse the next transaction and terminate on errors
+		var tx types.PoolTransaction
+		switch txType, err := stream.Uint(); txType {
+		case plainTxID:
+			tx = new(types.Transaction)
+		case stakingTxID:
+			tx = new(staking.StakingTransaction)
+		default:
+			if err != nil {
+				if err == io.EOF {
+					err = nil
```

Comment:
why ignore the EOF error? please comment.

Reply:
ignore EOF to exit reading from journal final without error, will document